istio-proxy/fetch.sh

```diff
@@ -2,28 +2,56 @@ set -x
 set -e
 
 PROXY_DIR=${PROXY_DIR:-istio-proxy}
+
 function check_envs() {
   if [ -z "$FETCH_DIR" ]; then
     echo "FETCH_DIR required. Please set"
     exit 1
   fi
 
-	PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
+  PROXY_FETCH_DIR=${FETCH_DIR}/${PROXY_DIR}
   CACHE_DIR=${PROXY_FETCH_DIR}/bazel
 }
 
+function check_git_hashes() {
+  local commit_hash_name="$1"
+  local tarball_hash_name="$2"
+  eval commit_hash=\$$commit_hash_name
+  eval tarball_hash=\$$tarball_hash_name
+
+  if [ ! -z "${commit_hash}" ] || [ ! -z "${tarball_hash}" ]; then
```

Comment:
I think you may to remove `!` from both conditionals

Reply:
I want to check to see if either is not empty. If one is empty but not the other then I want it to fail.

Reply:
if both are empty, the condition above will evaluate to `! true || ! true => false || false => false`. Is that what you want?

Reply:
Yes. If both are empty then I want to use the default (i.e. the current branch). That's ok. The purpose of that function is to error out if they set one and not the other as both are required to use a hash instead of the branch.